src/minichain/mempool.py

```
@@ -0,0 +1,247 @@
+"""Mempool data structures and transaction selection logic."""
+
+from __future__ import annotations
+
+import time
+from dataclasses import dataclass, field
+from heapq import heappop, heappush
+from typing import Iterable
```

Ruff (0.15.2)

[warning] Import from `collections.abc` instead: `Iterable`

Import from `collections.abc`

(UP035)